### PR TITLE
PackInvoker manages interactions with the tested pack CLI

### PR DESCRIPTION
**Pull request by simonjjones**
Signed-off-by: Simon Jones <simonjones@vmware.com>

## Summary
Following on from discussion in #673 this contributes further changes intended to improve the readability, maintainability and user experience of working with our acceptance test suite.

This change creates PackInvoker object which handles the task invoking `pack` commands throughout the acceptance suite, and a PackFixtureManager object which finds and templates fixtures for use in the tests.

There is a small change to the logic of which fixture is used for one test, which I believe uses the more appropriate source. Other than this, hopefully there are no changes to the logic of what is being tested, just rearrangement for clarity.

## Documentation

- Should this change be documented?
    - [ ] Yes, see #___
    - [x] No

## Related

Contributes to #606

### CODE REVIEW

File: acceptance/config/pack_assets.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// +build acceptance
+
+package config
+
+type PackAssetManager struct {
```
Comment (jromero) [inline acceptance/config/pack_assets.go:5]: Nitpick: `Manager` in my mind entails a service class when in reality this appears to simply be a data type.

```suggestion
type PackAsset struct {
```

Comment (simonjjones) [reply acceptance/config/pack_assets.go]: Renamed throughout

**Comment by simonjjones** [comment]
@jromero Should I be worried about the codecov report for this? I don't see how it can be right, as I haven't touched the unit tests, but it's alarmingly red, and don't want to ignore it out of hand.

**Comment by jromero** [comment]
@simonjjones it would be nice to know _why_ it changed although I agree with you that it shouldn't have.

**Comment by jromero** [comment]
Ok, so I think I've found the issue. It appears that both Windows jobs did not complete successfully on `main` on the latest commit. This in effect raised the _average_ code coverage percentage (aggregate of all build outcomes):


This PR (and any subsequent PRs) will lower the overall code coverage average as they properly report all build outcomes. Windows results effectively dragging the code coverage average down.

image

File: acceptance/acceptance_test.go [hunk 2]
```diff
@@ -799,22 +791,23 @@ func testAcceptance(
...
+
 							h.AssertEq(t, output, expectedOutput)
 						}
 					})
 
 					when("--no-color", func() {
 						it.Before(func() {
-							h.SkipIf(t,
-								packSemver.LessThan(semver.MustParse("0.12.0")) || !packSemver.Equal(semver.MustParse("0.0.0")),
+							h.SkipUnless(t,
```
Comment (dfreilich) [inline acceptance/acceptance_test.go:801]: I love this addition

File: acceptance/invoke/pack.go [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+func (i *PackInvoker) RunSuccessfully(name string, args ...string) string {
+	i.testObject.Helper()
+
+	output, err := i.Run(name, args...)
+	h.AssertNil(i.testObject, err)
+
+	return output
+}
+
+func (i *PackInvoker) JustRunSuccessfully(name string, args ...string) {
```
Comment (dfreilich) [inline acceptance/invoke/pack.go:101]: I appreciate this change, rather than the `PackWithOutput` pattern before. Thanks!

**Pull request merged**